hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/LiteralEndToEndTest.java

```diff
@@ -0,0 +1,109 @@
+import static com.hazelcast.sql.SqlColumnType.SMALLINT;
+import static com.hazelcast.sql.SqlColumnType.TINYINT;
+import static com.hazelcast.sql.SqlColumnType.VARCHAR;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class LiteralEndToEndTest extends ExpressionEndToEndTestBase {
+
+    @Test
+    public void testValid() {
+        assertRow("0", EXPR0, TINYINT, (byte) 0);
+        assertRow("-0", EXPR0, TINYINT, (byte) 0);
+        assertRow("000", EXPR0, TINYINT, (byte) 0);
+        assertRow("1", EXPR0, TINYINT, (byte) 1);
+        assertRow("-1", EXPR0, TINYINT, (byte) -1);
+        assertRow("-01", EXPR0, TINYINT, (byte) -1);
+        assertRow("001", EXPR0, TINYINT, (byte) 1);
+        assertRow("100", EXPR0, TINYINT, (byte) 100);
+        assertRow(Byte.toString(Byte.MAX_VALUE), EXPR0, TINYINT, Byte.MAX_VALUE);
+
+        assertRow(Short.toString((short) (Byte.MAX_VALUE + 1)), EXPR0, SMALLINT, (short) (Byte.MAX_VALUE + 1));
+        assertRow(Short.toString(Short.MAX_VALUE), EXPR0, SMALLINT, Short.MAX_VALUE);
+
+        assertRow(Integer.toString(Short.MAX_VALUE + 1), EXPR0, INTEGER, Short.MAX_VALUE + 1);
+        assertRow(Integer.toString(Integer.MAX_VALUE), EXPR0, INTEGER, Integer.MAX_VALUE);
+
+        assertRow(Long.toString(Integer.MAX_VALUE + 1L), EXPR0, BIGINT, Integer.MAX_VALUE + 1L);
+        assertRow(Long.toString(Long.MAX_VALUE), EXPR0, BIGINT, Long.MAX_VALUE);
+
+        assertRow("0.0", EXPR0, DECIMAL, new BigDecimal("0.0"));
+        assertRow("1.0", EXPR0, DECIMAL, new BigDecimal("1.0"));
+        assertRow("1.000", EXPR0, DECIMAL, new BigDecimal("1.000"));
+        assertRow("001.000", EXPR0, DECIMAL, new BigDecimal("1.000"));
+        assertRow("1.1", EXPR0, DECIMAL, new BigDecimal("1.1"));
+        assertRow("1.100", EXPR0, DECIMAL, new BigDecimal("1.100"));
+        assertRow("001.100", EXPR0, DECIMAL, new BigDecimal("1.100"));
+        assertRow("-0.0", EXPR0, DECIMAL, new BigDecimal("0.0"));
+        assertRow("-1.0", EXPR0, DECIMAL, new BigDecimal("-1.0"));
+        assertRow("-001.100", EXPR0, DECIMAL, new BigDecimal("-1.100"));
+        assertRow(".0", EXPR0, DECIMAL, BigDecimal.valueOf(0.0));
+        assertRow(".1", EXPR0, DECIMAL, BigDecimal.valueOf(0.1));
+
+        assertRow("0e0", EXPR0, DOUBLE, 0.0);
+        assertRow("1e0", EXPR0, DOUBLE, 1.0);
+        assertRow("1e000", EXPR0, DOUBLE, 1.0);
+        assertRow("001e000", EXPR0, DOUBLE, 1.0);
+        assertRow("1.1e0", EXPR0, DOUBLE, 1.1);
+        assertRow("1.100e0", EXPR0, DOUBLE, 1.1);
+        assertRow("001.100e0", EXPR0, DOUBLE, 1.1);
+        assertRow("-0.0e0", EXPR0, DOUBLE, 0.0);
+        assertRow("-1.0e0", EXPR0, DOUBLE, -1.0);
+        assertRow("-001.100e0", EXPR0, DOUBLE, -1.1);
+        assertRow(".0e0", EXPR0, DOUBLE, 0.0);
+        assertRow(".1e0", EXPR0, DOUBLE, 0.1);
+        assertRow("1.1e1", EXPR0, DOUBLE, 11.0);
+        assertRow("1.1e-1", EXPR0, DOUBLE, 0.11);
```

Comment:
Is it possible to specify `NaN` or infinite values as literal? AFAIK some databases, such as Postgres, can do that

Reply:
created an issue for 4.2 https://github.com/hazelcast/hazelcast/issues/17314